maven-builder-plugin/src/test/java/hudson/model/GetEnvironmentOutsideMavenBuildTest.java

```diff
@@ -0,0 +1,90 @@
+package hudson.model;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.io.IOException;
+
+import hudson.EnvVars;
+import hudson.matrix.MatrixProject;
+import hudson.maven.MavenModuleSet;
+import hudson.tasks.Maven.MavenInstallation;
+import hudson.util.StreamTaskListener;
+
+import jenkins.model.Jenkins;
+
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.ExtractResourceSCM;
+import org.jvnet.hudson.test.HudsonTestCase;
+import org.jvnet.hudson.test.ToolInstallations;
+
+/**
+ * Tests that getEnvironment() calls outside of builds are safe.
+ *
+ * @author kutzi
+ */
+@Issue("JENKINS-11592")
+public class GetEnvironmentOutsideMavenBuildTest extends HudsonTestCase {
```

Comment:
No, just an extraction of the maven specific methods on GetEnvironmentOutsideBuildTest. I tried no make no changes on the whole PR, with the exception of the new clases for the plugin, in order to reduce posible conflicts and make the review easier. A follow-up PR will include all the changes needed to actually remove the plugin from the project.